module/VuFind/src/VuFind/View/Helper/Root/Icon.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\Config\Config;
+use Laminas\View\Helper\AbstractHelper;
+use VuFind\Config\PluginManager as ConfigManager;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Icon extends AbstractHelper
+{
+    /**
+     * Icons config (icons.ini)
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Transforming map
+     *
+     * @var array
+     */
+    protected $iconMap;
+
+    /**
+     * Constructor
+     *
+     * @param Config $config Icons config (icons.ini)
+     */
+    public function __construct(Config $config)
+    {
+        $this->config = $config->Config;
+        $this->iconMap = $config[$config->Config->use];
```

Comment:
Might be worth having some defaults/fallbacks here, like:

<pre>
$this->config = $config->Config ?? new Config([]);
$this->iconMap = $config[$config->Config->use ?? 'FontAwesome'] ?? new Config([]);
</pre>

Reply:
Of course, I also wonder if the configuration belongs somewhere else entirely -- like in theme.config.php, since I could imagine we might want different icon systems in different themes.

Also, is it possible that picking one icon library isn't flexible enough? Would it make more sense to rearrange the configuration so that there's a default icon library, but you can override individual icons to use different systems?

Reply:
In consideration related to both of these points:
- icons.ini per theme (like languages per theme)
- `use=FontAwesome,Bootstrap,Tall Terry's Excellent Icons`

We could also look in `theme.config.php` for overrides but I thought the ini file would be cleaner and easier to edit.

Reply:
I'm not completely opposed to having an icons.ini per theme, but putting .ini files in themes would be a new thing we haven't done before. Having a section in themes.config.php would be more consistent with everything else we currently do in themes. You could likely expose the settings through the VuFindTheme\ResourceContainer.

I don't think a comma-separated list of icon sets is going to work, because how would the code know which one to use? I was thinking of something more like:

<pre>
[General]
defaultIconSet = FontAwesome

[Overrides]
favorite = Bootstrap
delete = "Tall Terry's Excellent Icons"
</pre>

So when we look up a particular icon, the request gets routed to the appropriate library. But maybe I'm making things more complicated than they need to be... just trying to aim for maximum flexibility.